OpenMTR/code/ReadData.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.IO;
+using System.Text.RegularExpressions;
+using OpenCvSharp;
+using System.Collections.Generic;
+
+
+namespace OpenMTR
+{
+    static class ReadData
+    {
+        // Get a list of data objects built from all the images found in a directory
+        public static List<DataObject> GetDataObjectList(string DirectoryPath)
+        {
+            List<DataObject> Data = new List<DataObject>();
+            foreach (string FilePath in Directory.GetFiles(DirectoryPath))
+            {
+                if (Regex.IsMatch(FilePath, ".(jpe?g|png)$"))
+                {
+                    Data.Add(GetDataObject(FilePath));
+                }
+            }
+            return Data;
+        }
+        public static void GetDataObjectList(string DirectoryPath, List<DataObject> Data)
+        {
+            Data = GetDataObjectList(DirectoryPath);
+        }
+
+        // Get a single data object from a path to the image file
+        public static DataObject GetDataObject(string ImagePath)
+        {
+            string Filename = Path.GetFileNameWithoutExtension(ImagePath);
+            int MetaReading = int.Parse(File.ReadAllLines(GetMetaPath(ImagePath))[0]);

```

Comment:
It would be safer and a little less code if you used `File.ReadAllText` vs `File.ReadAllLines`. Safer in the sense if _for some reason_ the number isn't on the first line.

Also, this should be in a try/catch block since there is a chance that `int.Parse()` could throw. I suggest if this throws `FormatException` or `ArgumentNullException`, we could set the `DataObject.MeterRead = -1` 
@cclinard84 Thoughts?